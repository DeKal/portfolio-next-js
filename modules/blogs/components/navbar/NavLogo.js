import React from 'react'
import styled from 'styled-components/macro'
import Navbar from 'react-bootstrap/Navbar'
import PropTypes from 'prop-types'

const NavLogo = ({ redirectLink, image }) => (
  <Container href={redirectLink}>
    <Logo alt="logo" src={image} />
  </Container>
)

export default NavLogo

NavLogo.propTypes = {
  redirectLink: PropTypes.string,
  image: PropTypes.string
}

const Container = styled(Navbar.Brand)`
  height: 50px;
  width: 125px;
  display: flex;
  margin-right: 0;
  overflow: hidden;
`

const Logo = styled.img`
  width: 100%;
  height: 250%;
  align-self: center;
`
