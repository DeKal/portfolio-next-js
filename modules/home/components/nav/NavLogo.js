import * as React from 'react'
import styled from 'styled-components/macro'

const NavLogo = () => (
  <Logo href="https://github.com/DeKal/portfolio-next-js">
    <LogoText>JD</LogoText>
  </Logo>
)

export default NavLogo

const Logo = styled.a`
  display: block;
  position: absolute;
  width: ${props => props.theme.navContentMinHeight};
  color: ${props => props.theme.colors.colorLight};
  font-size: ${props => props.theme.navHeaderHeight};
  line-height: ${props => props.theme.navContentMinHeight};
  font-family: ${props => props.theme.fonts.fontSerif};
  font-weight: bold;
  text-align: center;
  background-color: ${props => props.theme.colors.colorAccent};
  border: none;

  @media screen and (max-width: 700px) {
    display: none;
  }

  &::after {
    content: '';
    position: absolute;
    bottom: -29px;
    left: 0;
    z-index: 250;
    border-top: 30px solid ${props => props.theme.colors.colorAccent};
    border-left: 40px solid transparent;
    border-right: 40px solid transparent;
  }

  &:hover {
    color: ${props => props.theme.colors.colorDark};
    text-decoration: none;
  }

  &:active {
    color: ${props => props.theme.colors.colorDark};
  }
`

const LogoText = styled.span`
  position: relative;
  display: block;
  width: 100%;
  height: 100%;

  &::after {
    content: '';
    position: absolute;
    bottom: -15px;
    right: -2px;
    width: 50px;
    height: 5px;
    background-color: transparent;
    box-shadow: ${props => props.theme.extra.boxShadow};
    transform: rotate(-37deg);
  }

  &::before {
    content: '';
    position: absolute;
    bottom: -15px;
    left: -2px;
    width: 50px;
    height: 5px;
    background-color: transparent;
    box-shadow: ${props => props.theme.extra.boxShadow};
    transform: rotate(37deg);
  }
`
