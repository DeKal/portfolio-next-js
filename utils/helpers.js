import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
