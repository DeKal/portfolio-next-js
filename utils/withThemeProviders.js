import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
