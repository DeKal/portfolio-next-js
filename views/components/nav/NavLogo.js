import * as React from 'react'
import styled from 'styled-components/macro'

const NavLogo = () => (
  <Logo>
    <LogoText>JD</LogoText>
  </Logo>
)

export default NavLogo

const Logo = styled.a`
  display: block;
  position: absolute;
  top: 0;
  left: 20px;
  width: ${props => props.theme.navContentMinHeight};
  color: ${props => props.theme.colors.colorLight};
  font-size: 40px;
  line-height: ${props => props.theme.navContentMinHeight};
  font-family: ${props => props.theme.fonts.fontSans};
  letter-spacing: -5px;
  font-weight: bold;
  text-align: center;
  background-color: ${props => props.theme.colors.colorAccent};
  border: none;

  @media screen and (max-width: 960px) {
    display: none;
  }

  &::after {
    content: '';
    position: absolute;
    bottom: -29px;
    left: 0;
    z-index: 250;
    border-top: 30px solid ${props => props.theme.colors.colorAccent};
    border-left: 40px solid transparent;
    border-right: 40px solid transparent;
  }
`

const LogoText = styled.span`
  position: relative;
  display: block;
  width: 100%;
  height: 100%;

  &::after {
    content: '';
    position: absolute;
    bottom: -15px;
    right: -2px;
    width: 50px;
    height: 5px;
    background-color: transparent;
    box-shadow: 0 1px 2px -1px ${props => props.theme.colors.colorDark};
    transform: rotate(-37deg);
  }

  &::before {
    content: '';
    position: absolute;
    bottom: -15px;
    left: -2px;
    width: 50px;
    height: 5px;
    background-color: transparent;
    box-shadow: 0 1px 2px -1px ${props => props.theme.colors.colorDark};
    transform: rotate(37deg);
  }
`
