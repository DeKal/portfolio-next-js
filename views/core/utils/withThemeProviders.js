import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
