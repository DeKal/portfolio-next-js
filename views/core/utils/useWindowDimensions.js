import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
