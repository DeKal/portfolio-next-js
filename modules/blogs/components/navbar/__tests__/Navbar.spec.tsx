import * as React from 'react'
import { shallowWithTheme } from '~/modules/core/utils/withThemeProviders'
import NavLogo from '~/modules/blogs/components/navbar/NavLogo'

describe('Blog NavLogo', () => {
  it('should render without throwing an error', function () {
    const wrap = shallowWithTheme(<NavLogo />)
    expect(wrap).toMatchSnapshot()
  })
})
