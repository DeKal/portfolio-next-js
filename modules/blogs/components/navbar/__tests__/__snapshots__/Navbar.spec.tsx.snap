// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog NavLogo should render without throwing an error 1`] = `
<NavLogo__Container>
  <NavLogo__Logo
    alt="logo"
  />
</NavLogo__Container>
`;
